File: src/median.js

let input = prompt("Enter numbers separated by spaces:");


let arr = input.split(" ");
for (let i = 0; i < arr.length; i++) {
    arr[i] = Number(arr[i]);
}


for (let i = 0; i < arr.length; i++) {
    for (let j = 0; j < arr.length - i - 1; j++) {
        if (arr[j] > arr[j + 1]) {
  
            let temp = arr[j];
            arr[j] = arr[j + 1];
            arr[j + 1] = temp;
        }
    }
}


let median;
let n = arr.length;

if (n % 2 === 0) {
    median = (arr[n / 2 - 1] + arr[n / 2]) / 2;
} else {
    median = arr[Math.floor(n / 2)];
}


console.log("Sorted array:", arr);
console.log("Median:", median);

File: src/median.py
def find_median(numbers):
    if not numbers:
        return None 

    sorted_numbers = sorted(numbers)
    n = len(sorted_numbers)

    if n % 2 == 1: 
        median = sorted_numbers[n // 2]
    else: 
        mid1 = sorted_numbers[n // 2 - 1]
        mid2 = sorted_numbers[n // 2]
        median = (mid1 + mid2) / 2
    return median

print("Please enter numbers, separated by spaces (e.g., 10 20 30 40 50):")
user_input = input()

try:
    input_numbers = [float(x) for x in user_input.split()]
    if input_numbers:
        median_value = find_median(input_numbers)
        if median_value is not None:
            print(f"The median of the input numbers is: {median_value}")
        else:
            print("No numbers were entered to calculate the median.")
    else:
        print("No numbers were entered.")
except ValueError:
    print("Invalid input: Please enter valid numbers.")
except Exception as e:
    print(f"An error occurred: {e}")
